Avoid accessing arrays that are null or not set

Arevall opened the pull request:
Added some checks for arrays that can be null. This fixes some errors with PHP 7.4+

Commit:
Avoid accessing arrays that are null or not set

Added some checks for arrays that can be null. This fixes some errors
with PHP 7.4+

tessus commented on Mantis.php:
```diff
@@ -923,11 +923,11 @@ function renderMantis( $input, $args, $mwParser )
 
 		// create table header - use an array to specify which columns to display
-		if ($conf['header'])
+		if ($conf['header'] && isset($conf['show']) && is_array($conf['show'])) 
```
Are you sure this is necessary? `$conf['show']` has a default value that is an array.
I mean it doesn't hurt, just wondering.

Arevall replied:
You are definitely correct - I forgot to remove that check after debugging.  And yes, a whitespace did seems to sneak its way in. 

P.S. Thanks for the plugin :)

tessus replied:
> Would you prefer a new pull request with changes?

No, you can just add another commit.

I will have to add other checks after I merge this PR as well. I just procrastinated since I currently don't have a dev instance of MediaWiki and Mantis. So I will have to setup a VM...

tessus reviewed (commented):
Thanks, this is awesome. I have noticed at one point that there are warnings but forgot to fix them.

tessus commented:
Btw, when I look at the code there are a lot of other areas that need additional checks as well.
I think I will have to fix the entire code at one point. ;-)

tessus commented on Mantis.php:
```diff
@@ -923,11 +923,11 @@ function renderMantis( $input, $args, $mwParser )
 			foreach ($conf['show'] as $colname)
 			{
-				$header = ($conf['headername'][$colname] ? $conf['headername'][$colname] : ucfirst($colname));
+				$header = (isset($conf['headername']) && isset($conf['headername'][$colname])) ? $conf['headername'][$colname] : ucfirst($colname);
```
can you try the null coalescing operator? maybe that works as well.

Arevall replied:
Good idea. I replaced my changes with the null coalescing operator in a new commit.

tessus replied:
Awesome, thanks.

Commit:
Replaced some instances of isset() with Null coalescing operator

For brevity when null-check was needed in combination with a ternary
operation. The Null coalescing operator was added in PHP 7.
